servicedirectory/servicedirectory.rb

```diff
@@ -0,0 +1,304 @@
+# Copyright 2020 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Note: Code samples in this file set constants which cannot be set inside
+#       method definitions in Ruby. To allow for this, code snippets in this
+#       sample are wrapped in global lambdas.
+
+$create_namespace = lambda do |project:, location:, namespace:|
```

Comment:
could these be changed to methods `def create_namespace project:, location:, namespace:`?

Reply:
I followed the example here: https://github.com/GoogleCloudPlatform/ruby-docs-samples/blob/master/kms/kms.rb so that we could have the require and "ServiceDirectory  = " statements in each code sample in our docs page - I tried to do the same thing with functions but I got an error for redefining the constants.

Reply:
Looking at the additional comments this makes more sense, I will update these.